File: Kap13/Python/Listing39_47/Grafikprogramm.py
from MyGraphs import Dreieck, Kreis
from MyLogger import MyJsonLogger, MyPlainLogger


myD = Dreieck(0x000000, 0xff0000, 3.0, 2.0, 2.0)
logger1 = MyJsonLogger()
myD.registerLogger(logger1)

myK = Kreis(0x000000, 0xff0000, 3.0);
logger2 = MyPlainLogger()
myK.registerLogger(logger2)

print(myD.getUmfang())
print(myK.getUmfang())

File: Kap13/JavaScript/Listing39_47/Grafikprogramm.js
import Dreieck from "./mygraphs/Dreieck.mjs";
import Kreis from "./mygraphs/Kreis.mjs";
import MyJsonLogger from "./mylogger/MyJsonLogger.mjs";
import MyPlainLogger from "./mylogger/MyPlainLogger.mjs";

const myD = new Dreieck(0x000000, 0xff0000, 3.0, 2.0, 2.0);
const logger1 = new MyJsonLogger();
myD.registerLogger(logger1);

const myK = new Kreis(0x000000, 0xff0000, 3.0);
const logger2 = new MyPlainLogger();
myK.registerLogger(logger2);

console.log(myD.getUmfang());
console.log(myK.getUmfang());